container/go/cmd/pusher/pusher.go

```diff
@@ -74,8 +73,8 @@ func main() {
 	if *format == "docker" && filepath.Ext(*src) != ".tar" {
 		log.Fatalf("Invalid value for argument -src for -format=docker, got %q, want path to tarball file with extension .tar.", *src)
 	}
-	if *format == "legacy" && filepath.Base(*src) != configFile {
-		log.Fatalf("Invalid value for argument -src for -format=legacy, got %q, want path to %s", *src, configFile)
+	if *format == "legacy" && *configPath == "" {
```

Comment:
Also add errors for cases when options that shouldn't be specified are specified. example, configPath, legacyBaseImage & layers shouldn't be specified if format is not legacy.